frame/nomination-pools/src/lib.rs

```diff
@@ -582,25 +583,247 @@ pub struct PoolRoles<AccountId> {
+///
+/// An optional commission `change_rate` allows the pool to set strict limits to how much commission
+/// can change in each update, and how often updates can take place.
+#[derive(
+	Encode, Decode, DefaultNoBound, MaxEncodedLen, TypeInfo, DebugNoBound, PartialEq, Copy, Clone,
+)]
+#[codec(mel_bound(T: Config))]
+#[scale_info(skip_type_params(T))]
+pub struct Commission<T: Config> {
+	/// Optional commission rate of the pool along with the account commission is paid to.
+	pub current: Option<(Perbill, T::AccountId)>,
+	/// Optional maximum commission that can be set by the pool `root`. Once set, this value can
+	/// only be updated to a decreased value.
+	pub max: Option<Perbill>,
+	/// Optional configuration around how often commission can be updated, and when the last
+	/// commission update took place.
+	pub change_rate: Option<CommissionChangeRate<T::BlockNumber>>,
+	/// The block throttling should be checked from. This value will be updated on all commission
+	/// updates and when setting an initial `change_rate`.
+	pub throttle_from: Option<T::BlockNumber>,
+}
+
+impl<T: Config> Commission<T> {
+	/// Returns true if the current commission updating to `to` would exhaust the change rate
+	/// limits.
+	///
+	/// A commission update will be throttled (disallowed) if:
+	/// 1. not enough blocks have passed since the `throttle_from` block, if exists, or
+	/// 2. the new commission is greater than the maximum allowed increase.
+	fn throttling(&self, to: &Perbill) -> bool {
+		if let Some(t) = self.change_rate.as_ref() {
+			let commission_as_percent =
+				self.current.as_ref().map(|(x, _)| *x).unwrap_or(Perbill::zero());
+
+			// do not throttle if `to` is the same or a decrease in commission.
+			if *to <= commission_as_percent {
+				return false
+			}
+
+			// the attempted increase in commission relative to the current commission.
+			let attempted_increase = (*to).saturating_sub(commission_as_percent);
+
+			if self.throttle_from.map_or(
+				// Note: defensive only. `throttle_from` should always exist where `change_rate`
+				// has already been set, so this scenario should never happen.
+				//
+				// if no `throttle_from` exists. Therefore, the commission update is throttled via
+				// `max_increase` only.
+				attempted_increase > t.max_increase,
+				|f| {
+					// ensure the `max_increase` durations passed since the previous commission
+					// update allow the attempted commission increase.
+					//
+					// the total durations passed since the last commission update.
+					let blocks_passed = <frame_system::Pallet<T>>::block_number().saturating_sub(f);
+
+					// `min_delay` blocks must have been passed since `throttle_from`.
+					if blocks_passed < t.min_delay {
+						return true
+					}
+
+					// calculate intervals passed since `throttle_from`. If min delay is set to 0,
+					// set intervals passed to 1 as to only allow 1 * max_increase.
+					let intervals_passed = if t.min_delay == Zero::zero() {
```

Comment:
A `min_delay` of `0` allows a change rate of only a max increase, and vice-versa. Pool operators could opt to do this initially, and since change rates can only get more restrictive (or stay the same), they are free to increase the `min_delay` and maintain `max_increase` as and when they feel the need to. So I do believe the `0` value has its merit for the change rate.

Reply:
Okay sounds valid. I thought it was a storage invariant.